### ISSUE TITLE
Improve handling of fenced code blocks

### ISSUE THREAD

**Issue by H077y** [systemic-chaos/currently-fronting#179]
Heya! Didn't want to write this up as 3 separate issues since they're mostly related.

![image](https://user-images.githubusercontent.com/64114013/202005609-65ec3e35-6fea-49fe-a9a2-8082abb5074a.png)

Firstly, the code block should not have coloured text. Felix's description doesn't have any syntax name at the top of the code block to specify any, so it should just be white text.

Secondly, the code block clips behind the avatar. This happens on every profile with code block at the top.

Thirdly, the links are too long and escape the box.

If you're interested: https://front.plural.codes/udlri

**Comment by ArcanaOfSouls** [comment]
Hi~

Thanks for filing the issue! Please allow me to explain each of these in turn:

1. We configured the syntax highlighting for code blocks to be automatic (to allow a greater number of syntax-highlighted code blocks to be included without running up against the character limit). We hadn't considered the possibility of folks using it for non-code use cases without manually specifying `md` (for Markdown). Now that we're thinking about it more, our use of `md` for text blocks is probably atypical; and there are probably plenty of systems using code blocks purely for the purpose of them being monospaced. We'll adjust the parameter very soon (possibly even tonight) as an enhancement.
2. That's actually the expected functionality. Your system's avatars all have transparent backgrounds. The code block has to have a different colored background from the rest of the info to ensure legibility and delineation from non-code sections. There's no way to handle this on our end unless we fill transparent pixels on all avatars with a static color, which aside from being a resource-intensive operation (meaning pages would load more slowly), I suspect most users (ourselves included) want the transparency of backgrounds in their images preserved. There is an alternate view for the avatar that should eliminate the overlap by altering the postition of the avatar so that it is centered above the description instead of floating to the left. You can test it by viewing https://front.plural.codes/udlri?a=center. We have a few other query parameters as well. If you haven't seen them and are interested, [they're listed in the README](https://github.com/systemic-chaos/currently-fronting#query-parameters). We have some new ones planned, as well as an interface for configuring them. ([A roadmap](https://github.com/systemic-chaos/currently-fronting#query-parameters) is outlined in the README as well.)
3. Hm. I know what's causing that, and it is indeed a bug (caused by a CSS configuration). This *should* be easy to fix. In theory, all we'll need to do is add the `break-words` property to the `@apply` directive at [`styles/globals.css` on line 25](https://github.com/systemic-chaos/currently-fronting/blob/main/styles/globals.css#L25). We'll take care of it as soon as we can &mdash; hopefully at the same time as the enhancement for item 1.

**TL;DR**: If you're subscribed to this issue, you should receive updates as the changes for items 1. and 3. are pushed over the next few days. Item 2. is expected and intended functionality, but using a query parameter to configure the avatar placement should eliminate the observed behavior.

Glad you find our app useful!
Arcana

**Comment by ArcanaOfSouls** [comment]
@H077y

Heya!

Items 1. and 3. should be resolved by #181, which was just merged to main. Item 2. is expected and intended, but only because your avatars have transparent backgrounds and code blocks use a different background color from the body text. There is no simple way in CSS to automatically cut out a portion of a `div` that has certain content in front of it. However, we offer query parameters that will allow you rearrange your avatar on the page (to some extent), thus eliminating the overlap. Our post above has some ideas. :)

We're marking this one as closed and completed. Please feel free to reach out again with any suggestions for features or bug reports!

If for some reason the update doesn't resolve the issues you were experiencing, please do let us know by replying to this issue.

Thanks!
Arcana

**Comment by H077y** [comment]
Thanks for fixing the 1st and 3rd issue. However, I still feel like the second one _is_ an issue that's easy to fix.

Seeing as even though it's transparent, it would still show the background colour of the page, why not set the background colour of the image to the background? It does exactly the same as it does now, except the codeblock doesn't overflow. It's as simple as just adding `background-color:#1e293b` and fixes the overflowing issue without technically adding a background.
![image](https://user-images.githubusercontent.com/64114013/202237343-6f1d85a0-48c6-4428-80dd-569f01e6ae30.png)

**Comment by ArcanaOfSouls** [comment]
Hm. Interesting! That's a reasonable idea. Clearly we were thinking about this backwards.

Tailwind CSS will make it easy enough to implement for both dark and light modes via the necessary classes at the component level.

We've been exceedingly exhausted over the last few weeks for health reasons. That said, we can take care of this in the not-too-distant future (should be by the end of this week at the latest) when our health has improved.

We've reopened the issue in the meantime.

Thanks.

**Comment by H077y** [comment]
I'm glad I can help.

Don't stress yourselves, I hope you feel better soon /gen

**Comment by ArcanaOfSouls** [comment]
@H077y Whoops, I didn't mean to auto-close this without you getting a chance to take a look. And now PluralKit is temporarily down for maintenance (which also means the API is unavailable)... When PK is back online, please feel free to test again and let us know if you see the same issue (although it *should* be fixed).

**Comment by H077y** [comment]
@ArcanaOfSouls It looks like it's working great :D
Thanks for the fix 🖤 I hope you're feeling better
